### PR TITLE
ci(workflows): added GH Actions and removed TravisCI/Greenkeeper

### PR DESCRIPTION
**Pull request by Berkmann18**
One for semantic releases and one to take over the .travis.yml file

### CODE REVIEW

**Comment by Berkmann18** [comment]
@RichardLitt  For the actions to work, the repo needs to have a `GITHUB_TOKEN` and `NPM_TOKEN`. I can't see any Security > Secrets and variables section in the settings and running `gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/mntnr/name-your-contributors/actions/secrets` gave me an empty list of `secrets`. It seems like only you can set that up.

**Comment by RichardLitt** [comment]
Couldn't figure it out. Made you an admin. You should be able to, now!

**Comment by Berkmann18** [comment]
I added an `NPM_TOKEN` and a `GH_TOKEN` (in case the GH provided `GITHUB_TOKEN` doesn't have enough permissions). I noticed `greenkeeper-lockfile` still fails and it seems that based on what I read on https://www.npmjs.com/package/greenkeeper-lockfile, it doesn't support GH Actions (hopefully soon assuming the [issue I raised](https://github.com/greenkeeperio/greenkeeper-lockfile/issues/252) doesn't get ignored).

**Comment by RichardLitt** [comment]
Thank you, @Berkmann18! @gr2m - Any ideas why?

**Comment by gr2m** [comment]
Greenkeeper is no more. For a CLI I would replace greenkeeper with dependabot. Remove the badge from the README as well

Very cool to see you two work together 👋🏼

**Comment by RichardLitt** [comment]
That makes more sense! :) Thanks, @gr2m.

**Comment by Berkmann18** [comment]
Some integration tests are failing due to `GITHUB_TOKEN`; not sure what's going on there.

File: .github/workflows/semantic-release.yml [hunk 1]
```diff
@@ -13,10 +13,10 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v1
         with:
-          node-version: '12.x'
+          node-version: '20.x'
       - run: npm ci
       - name: Semantic Release
-        uses: cycjimmy/semantic-release-action@v2
+        uses: codfish/semantic-release-action@v2
```
Comment (gr2m) [inline .github/workflows/semantic-release.yml]: why the switch? Try bumping to `cycjimmy/semantic-release-action@v3`

**Comment by gr2m** [comment]
> Some integration tests are failing due to `GITHUB_TOKEN`; not sure what's going on there.

I don't see a token being passed in the workflow file. I don't think unauthorized requests to `POST /graphql` are allowed, hence the error.

**Comment by gr2m** [comment]
Try to pass it like this in `test.yml`

```yml
    - run: npm t
      env:
        CI: true
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by Berkmann18** [comment]
It looks like the CI works now 🥳.

File: .github/workflows/semantic-release.yml (hunk 1 above)
Comment (Berkmann18) [reply .github/workflows/semantic-release.yml]: Just saw this, `cycjimmy`'s one was a fork which led to issues while the action was running (I can't recall which ones now other than the fork being out-of-date and the initial version having bugfixes for the issue).

**Comment by Berkmann18** [comment]
I'll merge that, if there's any issues with the SR action, I'll try with `cycjimmy/semantic-release-action@v3`.

**Comment by RichardLitt** [comment]
\o/

**Pull request merged**